core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
I think its too much trouble: you could also do this:
```
private static final Random RANDOM = new Random(StringHelper.GOOD_FAST_HASH_SEED);
```

that one is already setup to respect `tests.seed`, otherwise its set to `System.currentTimeMillis`.